ztp/siteconfig-generator/siteConfig/siteConfig_test.go

```diff
@@ -88,21 +75,79 @@ spec:
     diskEncryption:
       type: nbde
   - clusterName: "defaults"
-    # Without further content under diskEncryptionthe type does not get populated
+    # Without further content under diskEncryption the type does not get populated
     diskEncryption:
       type:
   - clusterName: "explicit"
     diskEncryption:
       type: none
 `
 	siteConfig := SiteConfig{}
-	_ = yaml.Unmarshal([]byte(input), &siteConfig)
+	err := yaml.Unmarshal([]byte(input), &siteConfig)
+	assert.NoError(t, err)
 
-	// Validate ClusterType
+	// Validate Disk Encryption type
 	assert.Equal(t, siteConfig.Spec.Clusters[0].DiskEncryption.Type, "nbde")
 	assert.Equal(t, siteConfig.Spec.Clusters[1].DiskEncryption.Type, "none")
 	assert.Equal(t, siteConfig.Spec.Clusters[2].DiskEncryption.Type, "none")
+}
+
+func TestRoleCounters(t *testing.T) {
+	input := `
+apiVersion: ran.openshift.io/v1
+kind: SiteConfig
+spec:
+  clusters:
+  - clusterName: "ignore-user-supplied-values"
+    numMasters: 5
```

Comment:
if the user set num of masters and workers that mean he know what he is doing, we should check the nodes he set and raise error if the num of masters/workers didn't match.

Reply:
No, the user should never set numMasters or numWorkers.  The point of this change is that there is no reason to do so and we should remove that field from the CRD entirely.